src/main/kotlin/uk/gov/justice/digital/hmpps/personrecord/message/processor/PersonService.kt

```diff
@@ -0,0 +1,88 @@
+package uk.gov.justice.digital.hmpps.personrecord.message.processor
+
+import org.slf4j.LoggerFactory
+import org.springframework.stereotype.Service
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonAddressEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonAliasEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonContactEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.entity.PersonEntity
+import uk.gov.justice.digital.hmpps.personrecord.jpa.repository.PersonRepository
+import uk.gov.justice.digital.hmpps.personrecord.model.Person
+import uk.gov.justice.digital.hmpps.personrecord.model.types.SourceSystemType
+import uk.gov.justice.digital.hmpps.personrecord.service.TelemetryService
+import uk.gov.justice.digital.hmpps.personrecord.service.type.TelemetryEventType
+
+@Service
+class PersonService(
+  private val personRepository: PersonRepository,
+  private val telemetryService: TelemetryService,
+) {
+
+  fun processPerson(person: Person) {
+    validateCro(person)
+    val existingPersonEntity: PersonEntity? = getPersonEntityBySourceSystem(person)
+    handlePerson(existingPersonEntity, person)
+  }
+
+  private fun handlePerson(
+    existingPersonEntity: PersonEntity?,
+    person: Person,
+  ) {
+    if (existingPersonEntity != null) {
+      updatePersonEntity(person, existingPersonEntity)
+    } else {
+      createPersonEntity(person)
+    }
+  }
+
+  private fun getPersonEntityBySourceSystem(person: Person): PersonEntity? {
+    val existingPersonEntity: PersonEntity? = when (person.sourceSystemType) {
+      SourceSystemType.HMCTS -> person.defendantId?.let { personRepository.findByDefendantId(it) }
+      SourceSystemType.NOMIS -> TODO()
```

Comment:
This is the bit that I think we could try moving out of here. We know the source system before we get into this service.

Ideally this would just take a person and update it. It would be great if it know nothing of prison, probation or courts